src/kbmod/result_set.py

```diff
@@ -0,0 +1,521 @@
+import copy
+import csv
+import math
+import numpy as np
+import kbmod.search as kb
+
+class SharedTools:
+    """
+    This class manages tools that are shared by the classes Interface and
+    PostProcess.
+    
+    Legacy approach. Soon to be deprecated.
+    """
+
+    def __init__(self):
+        return
+
+    def gen_results_dict(self):
+        """
+        Return an empty results dictionary. All values needed for a results
+        dictionary should be added here. This dictionary gets passed into and
+        out of most Interface and PostProcess methods, getting altered and/or
+        replaced by filtering along the way.
+        """
+        keep = {
+            "stamps": [],
+            "new_lh": [],
+            "results": [],
+            "times": [],
+            "lc": [],
+            "lc_index": [],
+            "all_stamps": [],
+            "psi_curves": [],
+            "phi_curves": [],
+            "final_results": ...,
+        }
+        return keep
+
+
+class ResultDataRow:
+    """
+    This class stores a collection of related data from a single kbmod result.
+    """
+    __slots__ = ("_trajectory",
+                 "_stamp",
+                 "_final_likelihood",
+                 "_all_times",
+                 "_valid_indices",
+                 "_all_stamps",
+                 "_psi_curve",
+                 "_phi_curve",
+                 "_num_times",
+                )
+    
+    def __init__(self, trj, times):
+        self._trajectory = trj
+        self._stamp = None
+        self._final_likelihood = trj.lh
+        self._all_times = copy.copy(times)
+        self._valid_indices = [i for i in range(len(times))]
+        self._all_stamps = None
+        self._psi_curve = None
+        self._phi_curve = None
+        self._num_times = len(times)
+
+    @property
+    def trajectory(self):
+        return self._trajectory
+
+    @property
+    def stamp(self):
+        return self._stamp
+
+    @property
+    def final_lh(self):
+        return self._final_likelihood
+
+    @property
+    def final_likelihood(self):
+        return self._final_likelihood
+
+    @property
+    def times(self):
+        return self._all_times
+
+    @property
+    def valid_indices(self):
+        return self._valid_indices
+
+    @property
+    def all_stamps(self):
+        return self._all_stamps
+
+    @property
+    def psi_curve(self):
+        return self._psi_curve
+
+    @property
+    def phi_curve(self):
+        return self._phi_curve
```

Comment:
I'm fine with changing these to all be attributes if the overhead is going to be significant. I'm just trying to avoid the cases we run into with the "keep" dictionary where it is pretty easy to mess up the results. The effective setter/getter functions are an attempt to protect the user from this. But maybe it is sufficient to document.

Reply:
From afar having attributes here didn't look that dangerous to me, but I could just be unaware of some use-pattern in kbmod that makes that dangerous. If there is an use-case that really calls for us to be very explicit about it then this is the way to do it. Unfortunate, but what can one do.